lib/src/json/json_util.c

```diff
@@ -259,9 +259,11 @@ json_entity_t json_entity_new(enum json_value_e type, ...)
 		e = json_dict_new();
 		break;
 	case JSON_NULL_VALUE:
+		e = NULL;
```

Comment:
I would prefer if we init 'e' at the top of the function to shut up the static analysis tool, and remove this line and the break below so it falls through to the assert

Reply:
are we sure that logically json_null_value never occurs at this point in the code even from malformed input?